ichnaea/conf.py

```diff
@@ -0,0 +1,117 @@
+"""
+Application settings and configuration derived from the environment
+"""
+from __future__ import absolute_import
+
+import os
+import os.path
+
+from everett.component import ConfigOptions, RequiredConfigMixin
+from everett.manager import ConfigManager, ConfigOSEnv
+
+HERE = os.path.dirname(__file__)
+
+
+class AppConfig(RequiredConfigMixin):
+    required_config = ConfigOptions()
+    required_config.add_option(
+        'testing',
+        default='false',
+        parser=bool,
+        doc='Whether or not we are running tests.'
+    )
+
+    required_config.add_option(
+        'asset_bucket',
+        default='',
+        doc='???'
+    )
+    required_config.add_option(
+        'asset_url',
+        default='',
+        doc='???'
+    )
+
+    # Database related settings
+    required_config.add_option(
+        'db_library',
+        default='pymysql',
+        doc='one of "pymysql" or "mysqlconnector"'
+    )
+    required_config.add_option(
+        'db_readonly_uri',
+        doc=(
+            'uri for the readonly database; '
+            'mysql+LIBRARY://USER:PASSWORD@HOST:PORT/NAME'
+        )
+    )
+    required_config.add_option(
+        'db_readwrite_uri',
+        doc=(
+            'uri for the read-write database; '
+            'mysql+LIBRARY://USER:PASSWORD@HOST:PORT/NAME'
+        )
+    )
+    required_config.add_option(
+        'db_ddl_uri',
+        doc=(
+            'uri for the ddl database used for migrations; '
+            'mysql+LIBRARY://USER:PASSWORD@HOST:PORT/NAME'
+        )
+    )
+
+    required_config.add_option(
+        'sentry_dsn',
+        default='',
+        doc='Sentry DSN; leave blank to disable Sentry error reporting'
+    )
+
+    required_config.add_option(
+        'statsd_host',
+        default='',
+        doc='StatsD host; blank to disable StatsD'
+    )
+
+    required_config.add_option(
+        'redis_uri',
+        doc='uri for Redis; redis://HOST:PORT/DB'
+    )
+
+    required_config.add_option(
+        'mapbox_token',
+        default='',
+        doc=(
+            'Mapbox API key; if you do not provide this, then parts of '
+            'the site will not work'
+        )
+    )
+
+    required_config.add_option(
+        'geoip_path',
+        default=os.path.join(HERE, 'tests/data/GeoIP2-City-Test.mmdb'),
+        doc='absolute path to mmdb file for geoip lookups'
+    )
+
+    def __init__(self, config):
+        self.raw_config = config
+        self.config = config.with_options(self)
+
+    def __call__(self, *args, **kwargs):
+        return self.config(*args, **kwargs)
```

Comment:
I thought about doing getattr (or whatever it is) instead and supporting uppercase names, so then we could do something like this:

```
settings.TESTING
```

instead of this:

```
settings('testing')
```

If we do that, we'll break the getattr protocol since Everett will want to throw a different exception. That might be fine, though.

@jwhitlock Does the `settings.TESTING` syntax look better to you?

Reply:
As a Django user, ``settings.TESTING`` feels more familiar, but the syntax ``settings('testing')`` helps orient me that I'm not in Django, so let's keep it.